tests/main/views/test_drafts.py

```diff
@@ -56,6 +62,457 @@ def service_count(self):
+        res = self.client.put(
+            '/draft-services/copy-from/{}'.format(self.service_id),
+            data=json.dumps({
+                **self.updater_json,
+                **self.basic_questions_to_copy,
+                'targetFramework': 'g-cloud-7'
+            }),
+            content_type='application/json')
+
+        data = json.loads(res.get_data())
+        assert res.status_code == 201
+        assert data['services']['frameworkSlug'] == 'g-cloud-7'
+        assert 'serviceId' not in data['services']
+
+    def test_should_404_if_target_framework_does_not_exist(self):
+        res = self.client.put(
+            '/draft-services/copy-from/{}'.format(self.service_id),
+            data=json.dumps({
+                **self.updater_json,
+                **self.basic_questions_to_copy,
+                'targetFramework': 'z-cloud'
+            }),
+            content_type='application/json')
+
+        assert res.status_code == 404
+
+    def test_400_if_target_framework_and_not_questions_to_copy(self):
+        res = self.client.put(
+            '/draft-services/copy-from/0000000000',
+            data=json.dumps({
+                **self.updater_json,
+                'targetFramework': 'g-cloud-7'
+            }),
+            content_type='application/json',
+        )
+        assert res.status_code == 400
+        assert "Required data missing: 'questions_to_copy'" in json.loads(res.get_data(as_text=True))['error']
+
+    def test_existing_copied_draft_does_not_prevent_copy_to_new_framework(self):
+        res1 = self.client.put(
+            '/draft-services/copy-from/{}'.format(self.service_id),
+            data=json.dumps(self.updater_json),
+            content_type='application/json')
+
+        assert res1.status_code == 201
+
+        res2 = self.client.put(
+            '/draft-services/copy-from/{}'.format(self.service_id),
+            data=json.dumps({
+                **self.updater_json,
+                **self.basic_questions_to_copy,
+                'targetFramework': 'g-cloud-7'
+            }),
+            content_type='application/json')
+
+        assert res2.status_code == 201
+
+    def test_only_specified_questions_are_copied(self):
+        res = self.client.put(
+            '/draft-services/copy-from/{}'.format(self.service_id),
+            data=json.dumps({
+                **self.updater_json,
+                'targetFramework': 'g-cloud-7',
+                'questionsToCopy': ['serviceBenefits', 'termsAndConditionsDocumentURL']
+            }),
+            content_type='application/json')
+
+        # draft = json.loads(res.get_data(as_text=True))['services']
+        assert res.status_code == 201
+
+        draft = DraftService.query.filter(
+            DraftService.framework_id == 4,
+        ).first()
+
+        assert all(key in ['serviceBenefits', 'termsAndConditionsDocumentURL'] for key in draft.data.keys())
```

Comment:
Might be a bit cleaner doing a list/set comparison like `assert {'serviceBenefits', 'termsAndConditionalsDocumentURL'} == set(draft.data.keys())`